src/main/scala/sigmastate/eval/Evaluation.scala

```diff
@@ -404,6 +404,9 @@ trait Evaluation extends RuntimeCosting { IR =>
               out(th())
             } else
               out(false)
+          case ApplyBinOpLazy(op, In(x: Boolean), In(y)) if op == BinaryXorOp =>
+            val th = y.asInstanceOf[() => Boolean]
+            out(th() ^ x)
```

Comment:
@gostkin I'm not sure it should be lazy. Let's check how it is done in Scala, and do the same.